client/src/features/visualizer-threejs/store.ts

```diff
@@ -37,51 +43,59 @@ interface BlockStoreState {
 }
 
 export const useBlockStore = create<BlockStoreState>(set => ({
-    blocks: [],
-    blockOptions: {},
+    blocksToAdd: [],
+    blockColors: {},
+    blocksToScaleQueue: [],
     yPositions: {},
     zoom: ZOOM_DEFAULT,
     dimensions: { width: 0, height: 0 },
     isPlaying: false,
     bps: 0,
-
     addBlock: (newBlock, options) => {
         set(state => {
-            const prevBlockOptions = state.blockOptions[newBlock.id] || {};
+            const prevBlockColors = state.blockColors[newBlock.id] || {};
             return {
-                blocks: [...state.blocks, newBlock],
-                blockOptions: {
-                    ...state.blockOptions,
-                    [newBlock.id]: { ...prevBlockOptions, ...options }
+                blocksToAdd: [...state.blocksToAdd, newBlock],
+                blockColors: {
+                    ...state.blockColors,
+                    [newBlock.id]: { ...prevBlockColors, ...options }
                 }
             };
         });
     },
+    removeBlocks: (blockIds: string[]) => {
+        set(state => ({
+            ...state,
+            blocksToAdd: [...state.blocksToAdd.filter(b => !blockIds.includes(b.id))]
+        }));
+    },
     removeBlock: (blockId: string) => {
         set(state => {
-            const nextBlockOptions = { ...state.blockOptions };
-            delete nextBlockOptions[blockId];
+            const nextBlockColors = { ...state.blockColors };
+            delete nextBlockColors[blockId];
             return {
-                blocks: [...state.blocks.filter(b => b.id !== blockId)],
-                blockOptions: nextBlockOptions
+                blocksToAdd: [...state.blocksToAdd.filter(b => b.id !== blockId)],
+                blockOptions: nextBlockColors
             };
         });
     },
```

Comment:
> At this moment, remove block we don't use.
> Not sure if we need to keep it or better to remove.
I'll remove it.

> Technically this variable contains properties of node: like color. I guess we would like to store also some another params inside, like position, size. Because anyway we call it here
We already have `BlockState`, wdyt about moving this `color` property in there ?